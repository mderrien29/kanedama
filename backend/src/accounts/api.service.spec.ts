import { Test, TestingModule } from '@nestjs/testing';
import { ConfigModule } from '@nestjs/config';
import { of } from 'rxjs';

import { ApiService } from './api.service';

describe('ApiService', () => {
  let apiService: ApiService;
  const getMock = jest.fn(() => of({}));
  const httpServiceMock = { get: getMock };

  beforeEach(async () => {
    const app: TestingModule = await Test.createTestingModule({
      imports: [ConfigModule.forRoot({ envFilePath: '.env.test' })],
      providers: [
        ApiService,
        { provide: 'HttpService', useFactory: () => httpServiceMock },
      ],
    }).compile();

    apiService = app.get<ApiService>(ApiService);
  });

  describe('getAccounts', () => {
    it('should use the HttpService to call the expected url', async () => {
      await apiService.getAccounts();
      expect(getMock).toHaveBeenCalled();
      expect(getMock.mock.calls).toMatchSnapshot();
    });

    it('should return the result from the Api in a promise', async () => {
      const answer = await apiService.getAccounts();
      expect(answer).toMatchSnapshot();
    });
  });

  describe('getTransactions', () => {
    it('should use the HttpService to call the expected url', async () => {
      await apiService.getTransactions(
        'accountId',
        new Date('2020-08-30'),
        new Date('2020-01-01'),
      );
      expect(getMock).toHaveBeenCalled();
      expect(getMock.mock.calls).toMatchSnapshot();
    });

    it('should return the result from the Api in a promise', async () => {
      const answer = await apiService.getTransactions(
        'accountId',
        new Date('2020-08-30'),
        new Date('2020-01-01'),
      );
      expect(answer).toMatchSnapshot();
    });
  });

  describe('getOldestTransaction', () => {
    it('should use the HttpService to call the expected url', async () => {
      await apiService.getOldestTransaction('accountId');
      expect(getMock).toHaveBeenCalled();
      expect(getMock.mock.calls).toMatchSnapshot();
    });

    it('should return the result from the Api in a promise', async () => {
      const answer = await apiService.getOldestTransaction('accountId');
      expect(answer).toMatchSnapshot();
    });
  });
});
