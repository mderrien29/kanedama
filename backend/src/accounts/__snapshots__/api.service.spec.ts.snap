// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiService getAccounts should return the result from the Api in a promise 1`] = `"valueReturnedByApi"`;

exports[`ApiService getAccounts should use the HttpService to call the expected url 1`] = `
Array [
  Array [
    "https://kata.getmansa.com/accounts",
  ],
]
`;

exports[`ApiService getOldestTransaction should return the result from the Api in a promise 1`] = `"valueReturnedByApi"`;

exports[`ApiService getOldestTransaction should use the HttpService to call the expected url 1`] = `
Array [
  Array [
    "https://kata.getmansa.com/accounts/accountId/transactions",
  ],
]
`;

exports[`ApiService getTransactions should return the result from the Api in a promise 1`] = `"valueReturnedByApi"`;

exports[`ApiService getTransactions should use the HttpService to call the expected url 1`] = `
Array [
  Array [
    "https://kata.getmansa.com/accounts/accountId/transactions?from=2020-08-30T00%3A00%3A00.000Z&to=2021-01-01T00%3A00%3A00.000Z",
  ],
]
`;
